lib/docker/image.js

```diff
@@ -74,14 +74,17 @@ class Image {
       return string.template(template)(vars)
     }).filter(Boolean)
 
+    const registry = string.template(opts.registry)(vars);
+    const name = string.template(opts.name)(vars)
+
     const image = new(this)({
-      registry: opts.registry
+      registry
     , project: opts.project
-    , name: opts.name
+    , name
     , dockerfile: opts.dockerfile
     , build_id: opts.build
     , cwd: context.cwd
-    , tags: tags
+    , tags
```

Comment:
nit: would you mind keeping the longhand style consistent? `name: name`

Reply:
Would you mind adding something in the tests that illustrates that this works?

Reply:
👍 sure thing ill get on it as soon as i have a moment!